File: converted/utilities/tinyqueue/elide-tinyqueue.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-tinyqueue

This demonstrates calling the TypeScript tinyqueue implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One tinyqueue implementation shared across TypeScript and Python
- Consistent tiny and fast priority queue implementation across services
- No Python custom implementation needed
- Perfect for task scheduling, pathfinding, event processing
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# tinyqueue_module = require('./elide-tinyqueue.ts')

print("=== Python Calling TypeScript tinyqueue ===\n")

# Example: Basic usage
# result = tinyqueue_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses tiny and fast priority queue implementation")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for task scheduling, pathfinding, event processing")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide tinyqueue (TypeScript)         │")
print("│   conversions/tinyqueue/elide-tinyqueue.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same tiny and fast priority queue implementation everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  tinyqueue = require('./elide-tinyqueue.ts')")
print("  # Use it directly!")

File: converted/utilities/tinyqueue/elide-tinyqueue.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-tinyqueue
#
# This demonstrates calling the TypeScript tinyqueue implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One tinyqueue implementation shared across TypeScript and Ruby
# - Consistent tiny and fast priority queue implementation across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript tinyqueue ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# tinyqueue_module = Elide.require('./elide-tinyqueue.ts')

# Example: Basic usage
# result = tinyqueue_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses tiny and fast priority queue implementation"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for task scheduling, pathfinding, event processing"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide tinyqueue (TypeScript)         │"
puts "│   conversions/tinyqueue/elide-tinyqueue.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same tiny and fast priority queue implementation everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/tinyqueue.rb"
puts "  TINYQUEUE = Elide.require('./elide-tinyqueue.ts')"
